### PR TITLE
feat(catalog-backend): permit entity field subselection

### PR DESCRIPTION
**Pull request by freben**
Ran into the `set` lodash function while looking into something else, and got inspired.

### CODE REVIEW

File: plugins/catalog-backend/src/service/filterQuery.ts [hunk 1]
```diff
@@ -70,3 +73,44 @@ export function translateFilterQueryEntryToEntityFilters(
...
+
+  if (!fields.length) {
+    return x => x;
+  }
+
+  return input => {
+    const output: RecursivePartial<Entity> = {};
+
+    for (const field of fields) {
+      const value = lodash.get(input, field);
```
Comment (Rugvip) [inline plugins/catalog-backend/src/service/filterQuery.ts:108]: Would love to support something `jq` inspired like `relations[].type` for arrays, but could add that later. It's just wonky af to be able to use e.g. `relations[0].type`. Maybe we could just disallow the `[<n>]` syntax for now?

Thinking a few steps ahead I wanna be able to do something like a `expand=relations.type=owner&fields=relations[].entity.{kind,metadata.{name,namespace}}` type thing to expand relation entities. Not that exact syntax, but you get my point.

**Review by Rugvip — APPROVED** [review]
Yep, maybe not too needed at this point, but a common thing to have in REST APIs.

File: plugins/catalog-backend/src/service/filterQuery.ts (hunk 1 above)
Comment (freben) [reply plugins/catalog-backend/src/service/filterQuery.ts:114]: Yep agreed, I forbade array based filtering now, and added tests

**Pull request merged**